mt8195_adsp support

andyross opened the pull request:
Headers for the mt8195_adsp device.  See https://github.com/zephyrproject-rtos/zephyr/pull/61197 and https://github.com/zephyrproject-rtos/sdk-ng/pull/694

Note the disabled file (albeit a really obscure one I promise no one will need) in the HAL build.  Seems like current Cadence headers have skewed far enough away from our HAL code that some things are breaking.

andyross committed:
Add config headers for mt8195_adsp

Headers for the MT8195 Audio DSP.  Specifically these were extracted
from a just-configured Cadence RI-2021.6 toolchain for the core
"hifi4_8195_PROD".

Signed-off-by: Andy Ross <andyross@google.com>

dcpleung commented:
The `addi.a` is added by the `tie-asm.h` in your second commit inside `xchal_cp_AudioEngineLX_store`. If you change it to `addi`, there is no need to remove `state-asm.S`.

andyross committed:
soc/mt8195_adsp: Fix assembly for older toolchain

The "addi.a" instruction referenced in this file doesn't build with
non-Cadence assemblers.  By name, it would seem to be a "add immediate
single precision floating point", but the way it's used is clearly
as an integer counter in a regular GPR.  Use an addi, which seems to
be the intent, to make this build.  In practice this code won't ever
be executed by Zephyr apps anyway.

Signed-off-by: Andy Ross <andyross@google.com>

andyross commented:
Rebase and take Daniel's suggestion about the weird instruction.  Which... I mean I guess it's probably the same as an addi, but it's absolutely not documented anywhere...  But for sure building one possibly-broken code path in a HAL file that will never be used is cleaner than disabling the whole file.
